typescript/sdk/src/hook/MerkleRootInterceptorDeployer.ts

```diff
@@ -64,14 +68,22 @@ export class MerkleRootInterceptorDeployer extends HyperlaneDeployer<
   async deployIsmContracts(
     chain: ChainName,
     config: MultisigIsmConfig,
-  ): Promise<HyperlaneContracts<MerkleRootIsmFactories>> {
-    this.logger(`Deploying MerkleRootMultisigIsm to ${chain}`);
-    const ism = await this.ismFactory.deployMerkleRootMultisigIsm(
-      chain,
-      config,
-    );
-    return {
-      ism: ism,
-    };
+  ): Promise<HyperlaneContracts<MultisigIsmFactories>> {
+    const ism = await this.ismFactory.deploy(chain, config);
+    if (config.type === ModuleType.MERKLE_ROOT_MULTISIG) {
+      this.logger(`Deploying StaticMerkleRootMultisigIsm to ${chain}`);
+      return {
+        ism: ism as StaticMerkleRootMultisigIsm,
+      };
+    } else if (config.type === ModuleType.MESSAGE_ID_MULTISIG) {
+      this.logger(`Deploying StaticMessageIdMultisigIsm to ${chain}`);
+      return {
+        ism: ism as StaticMessageIdMultisigIsm,
+      };
+    } else {
+      throw new Error(
+        `Unexpected ISM type ${config.type} for MerkleRootInterceptorDeployer`,
+      );
+    }
```

Comment:
I guess this is the multichain config checking? still feels simplest to just directly reuse the ismFactory but 🤷‍♂️

Reply:
made it into a single `HyperlaneHookDeployer`  - https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2747/commits/a53dd91d02cc382612463742e19dcb8b278d1a73